addon/adapters/adapter.js

```diff
@@ -156,6 +173,46 @@ export default Adapter.extend(ImportExportMixin, {
     return null;
   },
 
+  _queryFilter(record, query = {}) {
+    return keys(query).every((key) => {
+      const value = query[key],
+        matches = key.match(/(.*)Id$/),
```

Comment:
one more question: do we go with the singular or plural:

```js
// singular
query('user', filter: { pets: { type: 'cat' } });

// plural
query('user', filter: { pets: { type: 'cats' } });
```

I guess the singular is more intuitive.